### ISSUE TITLE
Representation of particle mass versus density

### ISSUE THREAD

**Issue by jwise77** [enzo-project/enzo-e#83]
I'm opening up an issue to have a discussion on the particle masses. This was brought up in #49. The particles in the group `has_mass` will be included in the gravity solve. Currently, the mass field is defined to be density. This is how Enzo handled the particle masses to ease their mass deposition into grids for the gravity solve.

Do we keep this convention from Enzo?  Or do we rename the field to `particle_mass_density` or the like? Then we could have some derived field that contains their masses. I could be convinced either way.

**Comment by matthewturk** [comment]
This would somewhat reduce the complexity on the yt side.

**Comment by gregbryan** [comment]
I think we should break with Enzo and define mass as mass. It will just simplify things and make the code easier to read and modify.

**Comment by stefanarridge** [comment]
I agree @gregbryan. At the moment, as far as I can tell, the particle 'masses' are kept the same when blocks are refined or coarsened, which is not consistent with treating 'mass' as density. I believe that is the reason for these lines of code in [src/Enzo/enzo_EnzoMethodPmDeposit.cpp](https://github.com/enzo-project/enzo-e/blob/master/src/Enzo/enzo_EnzoMethodPmDeposit.cpp)
```
// Scale mass by volume if particle value is mass instead of density

      int level = block->level();

      // Required for Cosmology ("mass" is mass)
      // Not for Collapse ("mass" is density)
      if (cosmology) {
        dens *= std::pow(2.0,rank*level);
      }
```
This PmDeposit method would be simplified a lot by simply depositing the particle mass to the grid before dividing by the cell volume. One thing that I would like to clear up is the treatment of comoving vs proper quantities. There are lots of statements all over the code looking something like:

```
if (cosmology){
x *= cosmo_a;
}
```
I have generally found it quite confusing to keep track of whether quantities are comoving or proper. In this case, do we need to divide by the comoving or proper cell volume. I also discussed this in #85 with respect to the gravity time step. It might need to be the subject of a separate Issue.

```

**Comment by mabruzzo** [comment]
Resolved by PR #89
